src/utill.rs

```diff
@@ -638,9 +592,61 @@ pub(crate) fn verify_fidelity_checks(
     Ok(())
 }
 
+/// Tor Error grades
+#[derive(Debug)]
+pub enum TorError {
+    /// Io error
+    IO(std::io::Error),
+    /// Generic error
+    General(String),
+}
+
+impl From<std::io::Error> for TorError {
+    fn from(value: std::io::Error) -> Self {
+        TorError::IO(value)
+    }
+}
+
+#[cfg(not(feature = "integration-test"))]
+pub(crate) fn check_tor_status(control_port: u16, password: &str) -> Result<String, TorError> {
+    use std::io::BufRead;
+
+    let mut stream = TcpStream::connect(format!("127.0.0.1:{}", control_port))?;
+    let mut reader = BufReader::new(stream.try_clone()?);
+
+    let auth_command = format!("AUTHENTICATE \"{}\"\r\n", password);
+    stream.write_all(auth_command.as_bytes())?;
+
+    let mut response = String::new();
+    reader.read_line(&mut response)?;
+    if !response.starts_with("250") {
+        log::error!(
+            "Tor authentication failed: {}, please provide correct password",
+            response
+        );
+        return Err(TorError::General("Tor authentication failed".to_string()));
+    }
+
+    stream.write_all(b"GETINFO status/bootstrap-phase\r\n")?;
+    response.clear();
+    reader.read_line(&mut response)?;
+
+    if response.contains("PROGRESS=100") {
+        log::info!("Tor is fully started and operational!");
+    } else {
+        log::warn!("Tor is still starting, try again later: {}", response);
+    }
+
+    let hostname = fs::read_to_string("/var/lib/tor/coinswap/hostname")?;
```

Comment:
Whats your thought on using the app's data-dir for this path? default could be `~/.coinswap/maker/tor/`.  And also making the path configurable at the API.

This will allow multiple makers to run in the same system and still have different tor addresses.